Gemfile

```diff
@@ -21,7 +21,7 @@ group(:development, :test) do
   gem("smart_properties", require: false)
   gem("frozen_record", require: false)
   gem("sprockets", require: false)
-  gem("rails", require: false)
+  gem("rails", require: false) unless defined?(@specified_rails)
```

Comment:
I'm not convinced on the need to pollute the default Gemfile just for the sake of avoiding duplication in the gemfiles used for tests.

What do you think about creating an ERB template for the test gemfiles that could be parameterized based on the Rails version we want to test and the sorbet version parsed from the gemspec?

Reply:
The thing is that if we do that, we still end up duplicating the rest of the Gemfile, which leads to it becoming out of sync (as it currently is on `main`).

I based this approach off the approach taken by [Shopify/maintenance_tasks](https://github.com/Shopify/maintenance_tasks/tree/main/gemfiles), although I used instance variables differently due to differences in the arguments to `gem` (specifically, the presence of `require: false`).

Specifically with regards to using a template, that would mean we'd have to have a step in CI that renders the template before `bundle install` reads it, but that would necessarily mean doing it before installing any gems, which might be problematic.

Honestly, the cleanest solution would be if we were able to do something like

```ruby
eval_gemfile "../Gemfile" # Containing a regular gemfile

override do
  # No complaint about duplicate gem specification
  gem("rails", "~> 6.1.0", require: false)
end
```

But AFAIK Bundler doesn't support anything like that.

Reply:
IMO removing that duplication (and the chance for them to get out of sync) justifies the change here.

Reply:
Not saying this is better, but in [this project](https://github.com/getsentry/sentry-ruby/blob/master/sentry-rails/Gemfile) I use env var to control which version to use (from Rails 5 to main).

Pros:

- We only need 1 Gemfile and logics are concentrated
- No warnings AFAICT

Cons:

- If conditions